### PR TITLE
[MetricsAdvisor] Updated DataFeedSource constructors

### LINKED ISSUE #21616: [MetricsAdvisor] Revisit required parameters in Data Feed Sources
Some properties are required in the constructor, but not by the service. Check which ones we can remove from construction.

### PR DESCRIPTION
**Pull request by kinelski**
Adding/removing data source constructors based on what parameters are currently required by the service (fixes https://github.com/azure/azure-sdk-for-net/issues/21616).

### CODE REVIEW

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/CHANGELOG.md [hunk 1]
```diff
@@ -74,6 +78,7 @@
 - The class `DatasourceCredential` (now called `DataSourceCredentialEntity`) is now abstract.
 - `AlertQueryTimeMode` and `FeedbackQueryTimeMode` are now regular enums.
 - The enum `AuthenticationType` present in some `DataFeedSource` subtypes is now an extensible enum.
+- Removed constructor `(string workspaceId, string query)` from the `LogAnalyticsDataFeedSource`.
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/CHANGELOG.md:81]: Not actually supported by the service. Other properties are required now. We'll possibly add this constructor back again as new types of credentials are added.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSources/SqlServerDataFeedSource.cs [hunk 2]
```diff
@@ -17,14 +17,37 @@ public class SqlServerDataFeedSource : DataFeedSource
...
+        /// Initializes a new instance of the <see cref="SqlServerDataFeedSource"/> class. This constructor does not
+        /// set a <see cref="ConnectionString"/>, so you must assign an <see cref="AuthenticationType"/> to the
+        /// <see cref="Authentication"/> property. Currently, only the <see cref="AuthenticationType.SqlConnectionString"/>
+        /// authentication is supported without a connection string. If you intend to use another type of authentication,
+        /// see <see cref="SqlServerDataFeedSource(string, string)"/>.
+        /// </summary>
+        /// <param name="query">The query to retrieve the data to be ingested.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="query"/> is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="query"/> is empty.</exception>
+        public SqlServerDataFeedSource(string query)
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSources/SqlServerDataFeedSource.cs:29]: do u want to add a test for it? I think as is fine, but just in case u want to check query empty and the exception and etc...

**Review by maririos — APPROVED** [review]

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSources/SqlServerDataFeedSource.cs (hunk 2 above)
Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSources/SqlServerDataFeedSource.cs:29]: Added tests for new ctors.
